src/sentry/ratelimits/utils.py

```diff
@@ -121,7 +123,13 @@ def get_organization_id_from_token(token_id: str) -> int | None:
     from sentry.models import SentryAppInstallation
 
     installation = SentryAppInstallation.objects.get_by_api_token(token_id).first()
-    return installation.organization_id if installation else None
+
+    # Return a random uppercase/lowercase letter to avoid collisions caused by tokens not being
+    # associated with a SentryAppInstallation. This is a temporary fix while we solve the root cause
```

Comment:
is the problem that we're returning `None` from this function so the rate limit key has `None` in it?

Reply:
Yes that's what's causing the collisions